src/test/java/com/google/neighborgood/TaskServletTest.java

```diff
@@ -0,0 +1,172 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.neighborgood.servlets;
+
+import static com.google.appengine.api.datastore.FetchOptions.Builder.withLimit;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.appengine.tools.development.testing.LocalUserServiceTestConfig;
+import com.google.common.collect.ImmutableMap;
+import java.io.*;
+import javax.servlet.http.*;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/** Unit test on the TaskServlet file */
+@RunWith(JUnit4.class)
+public final class TaskServletTest {
+
+  /* Set up the test environment with Datastore and UserService and simulate the situation
+   * that the user is logged in as admin, has a specific email and userID.
+   */
+  private LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(
+              new LocalDatastoreServiceTestConfig(), new LocalUserServiceTestConfig())
+          .setEnvIsAdmin(true)
+          .setEnvIsLoggedIn(true)
+          .setEnvEmail("leonardzhang@google.com")
+          .setEnvAuthDomain("1234567890")
+          .setEnvAttributes(
+              ImmutableMap.of(
+                  "com.google.appengine.api.users.UserService.user_id_key", "1234567890"));
+
+  private UserService userService;
+  private DatastoreService ds;
+  private HttpServletRequest request;
+  private HttpServletResponse response;
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+    userService = UserServiceFactory.getUserService();
+    ds = DatastoreServiceFactory.getDatastoreService();
+    request = mock(HttpServletRequest.class);
+    response = mock(HttpServletResponse.class);
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void setUpTest() {
```

Comment:
This name is similar enough to `setUp` that it made me wonder if it was part of the JUnit framework; can we call it something like `helpersTest` or `testEnvironmentTest`?

...five minutes later I see that it has a `@Test` annotation, but it was still confusing to me :)

Reply:
Ah I think `testEnvironmentTest` is definitely a better name for the test case. I will change its name in the new commit. The current name is confusing 😭.